pkg/process/checks/format.go

```diff
@@ -0,0 +1,299 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+package checks
+
+import (
+	_ "embed"
+	"errors"
+	"io"
+	"sort"
+	"strconv"
+	"text/template"
+	"time"
+
+	model "github.com/DataDog/agent-payload/v5/process"
+	"github.com/DataDog/datadog-agent/pkg/process/config"
+
+	"github.com/dustin/go-humanize"
+)
+
+var (
+	// ErrNoHumanFormat is thrown when a check without human-readable support is passed to the HumanFormat method
+	ErrNoHumanFormat = errors.New("no implementation of human-readable output for this check")
+
+	//go:embed templates/processes.tmpl
+	processesTemplate string
+
+	//go:embed templates/rtprocesses.tmpl
+	rtProcessesTemplate string
+
+	//go:embed templates/containers.tmpl
+	containersTemplate string
+
+	//go:embed templates/rtcontainers.tmpl
+	rtContainersTemplate string
+
+	//go:embed templates/discovery.tmpl
+	discoveryTemplate string
+
+	fnMap = template.FuncMap{
+		"humanize": humanize.Commaf,
+		"bytes":    humanize.Bytes,
+		"time":     func(v int64) string { return time.UnixMilli(v).UTC().Format(time.RFC3339) },
+		"cpupct":   func(v float32) string { return humanize.Ftoa(float64(v)) + "%" },
+		"io": func(v float32) string {
+			if v < 0 {
+				return "-"
+			}
+			return humanize.Ftoa(float64(v))
+		},
+	}
+)
+
+// HumanFormat takes the messages produced by a check run and outputs them in a human-readable format
+func HumanFormat(check string, msgs []model.MessageBody, w io.Writer) error {
+	switch check {
+	case config.ProcessCheckName:
+		return humanFormatProcess(msgs, w)
+	case config.RTProcessCheckName:
+		return humanFormatRealTimeProcess(msgs, w)
+	case config.ContainerCheckName:
+		return humanFormatContainer(msgs, w)
+	case config.RTContainerCheckName:
+		return humanFormatRealTimeContainer(msgs, w)
+	case config.DiscoveryCheckName:
+		return humanFormatProcessDiscovery(msgs, w)
+	}
+	return ErrNoHumanFormat
+}
+
+func humanFormatProcess(msgs []model.MessageBody, w io.Writer) error {
+	var data struct {
+		Processes  []*model.Process
+		Containers []*model.Container
+		Hostname   string
+		CPUCount   int
+		Memory     uint64
+	}
+
+	var (
+		processes    = map[int32]*model.Process{}
+		containers   = map[string]*model.Container{}
+		pids         []int
+		containerIDs []string
+	)
+
+	for _, m := range msgs {
+		proc := m.(*model.CollectorProc)
```

Comment:
Should a cast check here to make sure we don't panic in case the msg is not of this type?

Reply:
Added a check here.